capstone/src/main/java/com/google/sps/servlets/UpdateVotesServlet.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.KeyFactory;
+
+@WebServlet("/update-votes")
+public class UpdateVotesServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get id of changed checkbox
+    String optionIdString = request.getParameter("id");
+    // Get whether checkbox is currently checked
+    String checked = request.getParameter("checked");
+    long optionId;
+
+    try {
+      optionId = Long.parseLong(optionIdString);
+    } catch (NumberFormatException e) {
+      System.err.println("ERROR: Failed to parse to long");
+      return;
+    }
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    // get userId of current logged in user
+    final String userId = "200";
+
+    // Get entity from datastore based on id
+    Entity optionEntity = null;
+    try {
+      optionEntity = datastore.get(KeyFactory.createKey("Option", optionId));
+    } catch (EntityNotFoundException e) {
+      System.err.println("ERROR: Could not get entity from datastore");
+      return;
+    }
+
+    /*
+     * Using ArrayList here because datastore will only return type ArrayList.
+     * Casting it to a HashSet will still have O(n) time complexity, so ArrayLists
+     * seem to be the best option in this scenario.
+     */
+    List<String> votes = (ArrayList<String>) optionEntity.getProperty("votes");
```

Comment:
I believe my comment above about the HashSet also applies here. While I agree that it would still have O(n) time complexity since you need to create the HashSet from the Map, that is an upfront cost that I believe is a good tradeoff since that would just occur once and then you can use the advantages of a Set, whereas with the List you'd be at its disadvantage while you use it